backend/src/apiserver/template/v2_template.go

```diff
@@ -107,37 +110,72 @@ func (t *V2Spec) GetTemplateType() TemplateType {
+		if decoder.Decode(&value) != nil {
+			break
+		}
+		valueBytes, err := goyaml.Marshal(value)
+		if err != nil {
+			return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, fmt.Sprintf("unable to marshal this yaml document: %s", err.Error()))
+		}
+		if isPipelineSpec(valueBytes) {
+			// Pick out the yaml document with pipeline spec
+			pipelineSpecJson, err := yaml.YAMLToJSON(valueBytes)
+			if err != nil {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, fmt.Sprintf("cannot convert v2 pipeline spec to json format: %s", err.Error()))
+			}
+			err = protojson.Unmarshal(pipelineSpecJson, &spec)
+			if err != nil {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, fmt.Sprintf("invalid v2 pipeline spec: %s", err.Error()))
+			}
+			if spec.GetPipelineInfo().GetName() == "" {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: name is empty")
+			}
+			match, _ := regexp.MatchString("[a-z0-9]([-a-z0-9]*[a-z0-9])?(\\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*", spec.GetPipelineInfo().GetName())
+			if !match {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: name should consist of lower case alphanumeric characters, '-' or '.', and must start and end with an alphanumeric character")
+			}
+			if spec.GetRoot() == nil {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: root component is empty")
+			}
+			v2Spec.spec = &spec
+		} else if isKubernetesExecutorConfig(valueBytes) {
+			// Pick out the yaml document with platform spec
+			var platformSpec pipelinespec.PlatformSpec
+			platformSpecJson, err := yaml.YAMLToJSON(valueBytes)
+			if err != nil {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPlatformSpec, fmt.Sprintf("cannot convert platform specific configs to json format: %s", err.Error()))
+			}
+			err = protojson.Unmarshal(platformSpecJson, &platformSpec)
+			if err != nil {
+				return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPlatformSpec, fmt.Sprintf("cannot unmarshal platform specific configs: %s", err.Error()))
+			}
+			v2Spec.platformSpec = &platformSpec
+		}
 	}
-	if spec.GetRoot() == nil {
-		return nil, util.NewInvalidInputErrorWithDetails(ErrorInvalidPipelineSpec, "invalid v2 pipeline spec: root component is empty")
-	}
-
-	return &V2Spec{spec: &spec}, nil
+	return &v2Spec, nil
 }
 
 func (t *V2Spec) Bytes() []byte {
 	if t == nil {
 		return nil
 	}
-	bytes, err := protojson.Marshal(t.spec)
+	bytesSpec, err := protojson.Marshal(t.spec)
+	if err != nil {
+		// this is unexpected, cannot convert proto message to JSON
+		return nil
+	}
+
+	bytesExecutorConfig, err := protojson.Marshal(t.platformSpec)
 	if err != nil {
 		// this is unexpected, cannot convert proto message to JSON
 		return nil
 	}
+	bytes := append(bytesSpec, bytesExecutorConfig...)
```

Comment:
To better understand how it works: if I take `string(bytes)`, would I get something

```
<YAML1>
---
<YAML2>
```

or 

```
 YAML1YAML2
```

Should we prefer the former over the later? WDYT?